File: Data Structures & Algorithms/first-missing-positive/submission-1.go
func firstMissingPositive(nums []int) int {
	n := len(nums)
	i := 0


	for i < n {
		// If the value at i is less than 0 or > n, skip
		if nums[i] <= 0 || nums[i] > n {
			i++
			continue
		}

		index := nums[i] - 1
		// if nums[i] and nums[index] don't match. swap them, otherwise continue
		if nums[i] != nums[index] {
			nums[i], nums[index] = nums[index], nums[i]
		} else {
			i++
		}
	}

	for i := 0; i < n; i++ {
		if nums[i] != i+1 {
			return i+1
		}
	}

	return n+1
}

File: Data Structures & Algorithms/concatenation-of-array/submission-1.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        vector<int> ans(2 * nums.size(), 0);
        for (int i = 0; i < ans.size(); i++) {
            int index = i % nums.size();
            ans[i] = nums[index];
        }

        return ans;
    }

};